projects/observability/src/pages/custom-dashboards/custom-dashboard-detail/custom-dashboard-detail.component.ts

```diff
@@ -0,0 +1,174 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { NavigationParamsType, NavigationService } from '@hypertrace/common';
+import { ButtonRole, InputAppearance, NotificationService } from '@hypertrace/components';
+import { Observable } from 'rxjs';
+import { CustomDashboardStoreService, DashboardData, PanelData } from '../custom-dashboard-store.service';
+import { CustomDashboardService } from '../custom-dashboard.service';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./custom-dashboard-detail.component.scss'],
+  template: `
+    <div class="dashboard-viewer">
+      <div class="header-container">
+        <ht-input
+          type="string"
+          class="dashboard-name-input"
+          appearance="${InputAppearance.Border}"
+          [value]="dashboardName"
+          (valueChange)="this.onDashboardNameChange($event)"
+        >
+        </ht-input>
+        <div class="button-container">
+          <ht-button
+            class="save-btn"
+            [label]="'Save'"
+            role="${ButtonRole.Additive}"
+            (click)="createOrUpdateDashboard()"
+          >
+          </ht-button>
+          <ht-button [label]="'Cancel'" role="${ButtonRole.Destructive}" (click)="redirectToListing()"> </ht-button>
+        </div>
+      </div>
+      <div class="panels-list" *ngIf="this.panels$ | async as panels">
+        <ht-custom-dashboard-panel
+          [panel]="panel"
+          *ngFor="let panel of panels"
+          (editPanel)="onPanelEdit($event)"
+          (deletePanel)="onPanelDelete($event)"
+        >
+        </ht-custom-dashboard-panel>
+      </div>
+      <div class="panel-container">
+        <div class="add-panel" (click)="redirectToCreatePanel()">Add Panel +</div>
+      </div>
+    </div>
+  `
+})
+export class CustomDashboardDetailComponent {
+  public dashboardName: string = 'Unnamed';
+  public isNew: boolean = false;
+  public dashboardId: string = '';
+  public dashboardData!: DashboardData;
+  public unSaved: boolean = false;
+  public queryParams: Params = {};
+  public panels$!: Observable<PanelData[]>;
+  public constructor(
+    protected readonly navigationService: NavigationService,
+    protected readonly customDashboardStoreService: CustomDashboardStoreService,
+    protected readonly activedRoute: ActivatedRoute,
+    private readonly customDashboardService: CustomDashboardService,
+    private readonly notificationService: NotificationService,
+    public readonly cdRef: ChangeDetectorRef
```

Comment:
```suggestion
    public readonly changeDetectorRef: ChangeDetectorRef
```

Abbreviations make the codebase hard to read in the long run.